python/lsst/daf/butler/cli/utils.py

```diff
@@ -689,8 +689,64 @@ def __init__(self, *args, **kwargs):
             kwargs["callback"] = catch_and_exit(callback)
         super().__init__(*args, **kwargs)
 
+    def _capture_args(self, ctx, args):
```

Comment:
Capturing the args is not new behavior; captured args are used by pipetask to turn CLI options into pipeline actions. There are 5 different options that result in a pipeline action (--task, --delete, --config, --config-file, --instrument), and the order these options are used on the command line specifies the order of the actions to be run. 

Click doesn't include any facility get the order of different options applied, it's a non-standard use case I think.

The problem with the way we were capturing arguments before was that they were not parsed enough; they were separated by whitespace, but not by separators ("="). And there's conditions where the separator should be included or excluded from the value. So I wrote this helper function to parse the list of CLI arguments, and reconstruct the list of option names & values, and argument values from the result.

I'll add some comments to the code and to the commit message.

Reply:
Thanks. That explanation helps a lot because it wasn't clear that this was specifically a pipetask-needing-oddly-specific-behavior problem versus a more general click-parsing-options problem.